### ISSUE TITLE
ERROR io.goobox.sync.sia.App - Failed to communicate SIA daemon: 404 page not found

### ISSUE THREAD

**Issue by MeijeSibbel** [GooBox/goobox-sync-sia#20]
This closes the process to terminate. V0.0.9. Windows 10 Home, 64 bit.

```
C:\Users\meije\Downloads\goobox-sync-sia-0.0.9-win64\goobox-sync-sia-0.0.9-win64\bin>goobox-sync-sia.bat
2017/12/11 00:05:42 INFO  io.goobox.sync.sia.Config - Loading config file C:\Users\meije\AppData\Local\Goobox\goobox.properties
2017/12/11 00:05:42 WARN  io.goobox.sync.sia.Config - Invalid parity pieces 0, minimum 12 pieces are required
2017/12/11 00:05:42 INFO  io.goobox.sync.sia.Config - Sync directory: C:\Users\meije\Goobox
2017/12/11 00:05:42 INFO  io.goobox.sync.sia.Config - Sync configuration: data pieces = 0, parity pieces = 12
2017/12/11 00:05:42 INFO  io.goobox.sync.sia.App - Checking if local Goobox sync folder exists: C:\Users\meije\Goobox
2017/12/11 00:05:42 INFO  io.goobox.sync.sia.App - Checking if Goobox data folder exists: C:\Users\meije\AppData\Local\Goobox
2017/12/11 00:05:44 INFO  io.goobox.sync.sia.App - Waiting SIA daemon starts
2017/12/11 00:05:44 INFO  io.goobox.sync.sia.SiaDaemon - Starting SIA daemon
2017/12/11 00:05:44 DEBUG io.goobox.sync.sia.SiaDaemon - Execute: C:\Users\meije\Downloads\goobox-sync-sia-0.0.9-win64\goobox-sync-sia-0.0.9-win64\sia\siad.exe --api-addr=127.0.0.1:9980 --host-addr=:9982 --rpc-addr=:9981 --sia-directory=C:\Users\meije\AppData\Local\Goobox\sia --modules=cgrtw
2017/12/11 00:05:44 DEBUG io.goobox.sync.sia.SiaDaemon - Loading...
2017/12/11 00:05:44 DEBUG io.goobox.sync.sia.SiaDaemon - (0/5) Loading siad...
2017/12/11 00:05:44 DEBUG io.goobox.sync.sia.SiaDaemon - (1/5) Loading gateway...
2017/12/11 00:05:44 DEBUG io.goobox.sync.sia.SiaDaemon - (2/5) Loading consensus...
2017/12/11 00:05:44 DEBUG io.goobox.sync.sia.SiaDaemon - (3/5) Loading transaction pool...
2017/12/11 00:05:44 DEBUG io.goobox.sync.sia.SiaDaemon - (4/5) Loading wallet...
2017/12/11 00:05:44 DEBUG io.goobox.sync.sia.SiaDaemon - (5/5) Loading renter...
2017/12/11 00:06:44 ERROR io.goobox.sync.sia.App - Failed to communicate SIA daemon: 404 page not found

2017/12/11 00:06:44 INFO  io.goobox.sync.sia.SiaDaemon - Closing SIA daemon

C:\Users\meije\Downloads\goobox-sync-sia-0.0.9-win64\goobox-sync-sia-0.0.9-win64\bin>
```

**Comment by MeijeSibbel** [comment]
If i start siad.exe from a different CMD window and then run the .bat file i get this output:

```
2017/12/11 00:12:15 INFO  io.goobox.sync.sia.Config - Loading config file C:\Users\meije\AppData\Local\Goobox\goobox.properties
2017/12/11 00:12:15 WARN  io.goobox.sync.sia.Config - Invalid parity pieces 0, minimum 12 pieces are required
2017/12/11 00:12:15 INFO  io.goobox.sync.sia.Config - Sync directory: C:\Users\meije\Goobox
2017/12/11 00:12:15 INFO  io.goobox.sync.sia.Config - Sync configuration: data pieces = 0, parity pieces = 12
2017/12/11 00:12:15 INFO  io.goobox.sync.sia.App - Checking if local Goobox sync folder exists: C:\Users\meije\Goobox
2017/12/11 00:12:15 INFO  io.goobox.sync.sia.App - Checking if Goobox data folder exists: C:\Users\meije\AppData\Local\Goobox
2017/12/11 00:12:15 INFO  io.goobox.sync.sia.App - Unlocking a wallet
2017/12/11 00:12:15 INFO  io.goobox.sync.sia.App - Failed to unlock the wallet: error when calling /wallet/unlock: wallet has not been encrypted yet
2017/12/11 00:12:15 INFO  io.goobox.sync.sia.App - Checking consensus DB
2017/12/11 00:12:15 INFO  io.goobox.sync.sia.App - Consensus DB isn't synchronized, wait a minute
```
```
C:\Users\meije\Downloads\goobox-sync-sia-0.0.9-win64\goobox-sync-sia-0.0.9-win64\sia>siad.exe
Loading...
(0/6) Loading siad...
(1/6) Loading gateway...
(2/6) Loading consensus...
(3/6) Loading transaction pool...
(4/6) Loading wallet...
(5/6) Loading host...
(6/6) Loading renter...
Finished loading in 0.0977329 seconds
```

**Comment by jkawamoto** [comment]
This is because the app doesn't wait enough until SIA daemon boots. It's solved in f2369486441085d793d0f6afef3ee4fa1fd14e60.
